### ISSUE TITLE
Indexed vector count is 0

### ISSUE THREAD

**Issue by ajainvivek** [qdrant/qdrant#2659]
Hi Team,

I have created collection with below configuration

```
{
  "result": {
    "status": "green",
    "optimizer_status": "ok",
    "vectors_count": 10373,
    "indexed_vectors_count": 0,
    "points_count": 10373,
    "segments_count": 10,
    "config": {
      "params": {
        "vectors": {
          "size": 1536,
          "distance": "Cosine"
        },
        "shard_number": 2,
        "replication_factor": 2,
        "write_consistency_factor": 1,
        "on_disk_payload": false
      },
      "hnsw_config": {
        "m": 16,
        "ef_construct": 100,
        "full_scan_threshold": 10000,
        "max_indexing_threads": 0,
        "on_disk": false
      },
      "optimizer_config": {
        "deleted_threshold": 0.2,
        "vacuum_min_vector_number": 1000,
        "default_segment_number": 8,
        "max_segment_size": 800000,
        "memmap_threshold": 800000,
        "indexing_threshold": 20000,
        "flush_interval_sec": 5,
        "max_optimization_threads": 1
      },
      "wal_config": {
        "wal_capacity_mb": 32,
        "wal_segments_ahead": 0
      },
      "quantization_config": {
        "binary": {
          "always_ram": true
        }
      }
    },
    "payload_schema": {}
  },
  "status": "ok",
  "time": 0.001475618
}
```

indexed_vectors_count value is 0 

Not sure what i am doing wrong here

Can someone help please

**Comment by timvisee** [comment]
Hi @ajainvivek!

Don't worry. This is expected behavior.

If the number of vectors is relatively low, no index is used. That is, because for a low number of vectors, it is quicker to search through the vectors directly rather than searching through an index.

We have `indexing_threshold` to configure this. As you can see in your above snippet, the threshold is at 20k by default. That means that if the number of vectors (per segment) gets larger than this threshold, the index is created automatically.

You could lower this threshold, but I don't think there's a good reason to do so.

So, in other words: if you start adding more vectors, the index will be created eventually automatically.

Thank you for sharing your collection information right away, that is very useful.

**Comment by ajainvivek** [comment]
@timvisee Thanks for the info. After lowering the `indexing_threshold` to 4k, I can see the `indexed_vectors_count` to set 10373. Now the results are empty in the after search. 

PATCH collections/334d3d0f-eb27-472a-893d-9246b8216c4a
{
  "optimizer_config": {
    "deleted_threshold": 0.2,
    "vacuum_min_vector_number": 1000,
    "default_segment_number": 8,
    "max_segment_size": 800000,
    "memmap_threshold": 800000,
    "indexing_threshold": 4000,
    "flush_interval_sec": 5,
    "max_optimization_threads": 1
  }
}

**Comment by timvisee** [comment]
Good, then everything is indexed!

What query did you use that now gives zero results?

Again, I actually recommend to keep the threshold at at least 20k. I'm not sure how healthy index (HNSW) searches are on low counts.

**Comment by ajainvivek** [comment]
@timvisee I realised the scores returned are above 1 for eg: "score": 894, "score": 780 etc

I assumed the scores are below 1.

The search was taking around ~4-5 secs. That is why i trying to check if indexing improves the performance.

**Comment by ajainvivek** [comment]
> We don't have any caching layers. Though, the HNSW index may give back very similar (or even the same) results.

The search query accuracy has reduced drastically and behaviour seems consistent for subsequent searches with different queries.

> Just out of curiosity. What are the search timings with and without the index?

Without indexing avg response time ~3-5 seconds
With indexing the avg response time ~2-5 seconds on varied requests

**Comment by timvisee** [comment]
Would you mind to share your exact queries?

**Comment by ajainvivek** [comment]
Sure
```
const queryData: any = {
      vector: embeddings[0].vector,
      filter: {
        must: [
          {
            key: 'docId',
            match: {
              any: docIds,
            },
          },
          {
            key: 'chunkSize',
            match: {
              any: chunkSize ? chunkSize : [1000],
            },
          },
        ],
      },
      limit,
    };
    
this.client.search(collectionName, queryData);
```
